src/grants/client_credentials.grant.ts

```diff
@@ -30,12 +30,44 @@ export class ClientCredentialsGrant extends AbstractGrant {
   }
 
   async respondToIntrospectRequest(req: RequestInterface): Promise<ResponseInterface> {
-    // introspection is authenticated via client_credentials, but we don't want to require this param in the request
-    req.body["grant_type"] = "client_credentials";
+    const { parsedToken, oauthToken, expiresAt, tokenType } = await this.tokenFromRequest(req);
+
+    const active = expiresAt > new Date();
+
+    const responseBody: OAuthTokenIntrospectionResponse = active
+      ? {
+          active: true,
+          scope: oauthToken.scopes.map(s => s.name).join(this.options.scopeDelimiter),
+          client_id: oauthToken.client.id,
+          token_type: tokenType,
+          ...parsedToken,
+        }
+      : { active: false };
+
+    const response = new OAuthResponse();
+    response.body = responseBody;
+    return response;
+  }
+
+  canRespondToRevokeRequest(request: RequestInterface): boolean {
+    return this.getRequestParameter("token_type_hint", request) !== "auth_code";
+  }
+
+  async respondToRevokeRequest(req: RequestInterface): Promise<ResponseInterface> {
+    let { oauthToken } = await this.tokenFromRequest(req);
+
+    await this.tokenRepository.revoke(oauthToken);
+
+    return new OAuthResponse();
+  }
+
+  private async tokenFromRequest(req: RequestInterface) {
+    req.body["grant_type"] = this.identifier;
```

Comment:
this means that in order to use access_token and refresh_token revoke, the authorization server needs to enable the `client_credentials` grant. For the record, the same goes for the introspection

Reply:
I've been thinking this for a while, but I've decided that the authorization server should have client_credentials and refresh_token grants enabled by default.